Revit_Core_Engine/Create/View/ViewPlan.cs

```diff
@@ -97,6 +97,107 @@ public static View ViewPlan(this Document document, Autodesk.Revit.DB.Level leve
+        [Input("cropRegionVisible", "True if the crop region should be visible.")]
+        [Input("annotationCrop", "True if the annotation crop should be visible.")]
+        [Output("viewPlan", "The new view.")]
+        public static View ViewPlan(this Document document, Level level, string viewName, ViewFamily viewFamily = ViewFamily.FloorPlan,  ElementId scopeBoxId = null, ElementId viewTemplateId = null, bool cropRegionVisible = false, bool annotationCrop = false)
+        {
+            View result = null;
+
+            if (!(viewFamily == ViewFamily.FloorPlan || viewFamily == ViewFamily.CeilingPlan || viewFamily == ViewFamily.AreaPlan || viewFamily == ViewFamily.StructuralPlan))
+            {
+                BH.Engine.Base.Compute.RecordWarning("Could not create View of type " + viewFamily + "'." + ". It has to be a FloorPlan, CeilingPlan, AreaPlan, or StructuralPlan ViewType.");
+                return result;
+            }
+
+            ViewFamilyType viewFamilyType = Query.ViewFamilyType(document, viewFamily);
+
+            result = Autodesk.Revit.DB.ViewPlan.Create(document, viewFamilyType.Id, level.Id);
+
+            if (viewTemplateId != null)
+            {
+                if (!(document.GetElement(viewTemplateId) as View).IsTemplate)
+                {
+                    BH.Engine.Base.Compute.RecordWarning("Could not apply the View Template of Id " + viewTemplateId + "'." + ". Please check if it's a valid View Template.");
+                    return result;
+                }
+
+                try
+                {
+                    result.ViewTemplateId = viewTemplateId;
+                }
+                catch (Exception)
+                {
+                    BH.Engine.Base.Compute.RecordWarning("Could not apply the View Template of Id " + viewTemplateId + "'." + ". Please check if it's a valid ElementId.");
+                }
+            }
+
+            if (!string.IsNullOrEmpty(viewName))
+            {
+                int number = 0;
+
+                while (true)
+                {
+                    try
+                    {
+                        if (number != 0)
+                        {
+#if (REVIT2018 || REVIT2019)
+                            result.ViewName = $"{viewName} ({number})";
+#else
+                            result.Name = $"{viewName} ({number})";
+#endif
+                            BH.Engine.Base.Compute.RecordWarning("There is already a view named '" + viewName + "'." + " It has been named '" + result.Name + "' instead.");
+                            break;
+                        }
+
+#if (REVIT2018 || REVIT2019)
+                        result.ViewName = viewName;
+#else
+                        result.Name = viewName;
+#endif
+                        break;
+                    }
+                    catch
+                    {
+                        number++;
+                    }
+                }
```

Comment:
To avoid checking the Revit version before setting the view name, you can do `result.get_Parameter(BuiltInParameter.VIEW_NAME).Set(newName);`

Also, similar to sheet number, it would be nice to avoid the `while(true)` and `try catch` with something like this:
```c#
int number = 0;
string uniqueName = viewName;

while (uniqueName.IsExistingViewName())
{
    number++;
    uniqueName = $"{viewName} ({number})";
}

result.get_Parameter(BuiltInParameter.VIEW_NAME).Set(uniqueName);
if (uniqueName != viewName)
{
    BH.Engine.Base.Compute.RecordWarning($"There is already a view named '{viewName}'. It has been named '{uniqueName}' instead.");
}
```